murdock/tests/test_notify.py

```diff
@@ -31,6 +32,14 @@
 )
 
 
+# Flush the prometheus collector registry between tests
```

Comment:
> Why is this required here ? It's also duplicated code from test_murdock.py. Is it possible to define it in a conf_test.py file ? (without autouse preferably, if possible)

Because the prometheus collector module keeps a global with all registered metrics. It trows an exception if you try to register a single metric multiple times. This flushes the registry between tests so that it doesn't throw an error at the second (and next) test.

I think it can be moved to the `conftest.py`.

Reply:
+1 for moving it to contest.py. I don't remember if there's one already.

Reply:
There is one already. What's the best way to add the fixture to the two modules? Do I have to add it to every relevant function or is there a way to mark it as used by all the functions in the file?

Reply:
You could keep `autouse=True` but that's a bit overkill. The best way is indeed to use it with each test function (by adding it as a parameter). I don't how many are impacted.

Maybe it's possible to autouse it globally but don't do anything for unrelated test modules. Maybe this doc could help: https://docs.pytest.org/en/6.2.x/fixture.html#fixtures-can-introspect-the-requesting-test-context (define some `require_prometheus = True` global variable in the test module and do something in the fixture if it's not None, or something like that. It's a bit hacky but should work)

Reply:
> I don't how many are impacted.

Every test creating a `Murdock` instance.

If there is a way to patch the `Murdock` creator with a clear of the registry during testing that would have my preference.

The other solution I've found is to wrap all tests in `test_murdock.py` in a class and add the usefixture decorator to the class. But that's going to create a huge diff here :)

Reply:
Then how about adding a fixture that yields a Murdock instance and depends on the prometheus registry fixture ? The `murdock` fixture would be added as a parameter to each test function that needs one.

Reply:
Implemented this. As this PR got to contain a bit more than I initially expected, let me know if you want me to split of af7ec35 to a separate PR

Reply:
Looks good. Just a missing newline in the conftest.py file. Please squash directly

Reply:
Ok, just give me a bit to make the CI happy again